### ISSUE TITLE
ExpressionAttributeNames/Values can include hyphens

### ISSUE THREAD

**Issue by metuuu** [tywalch/electrodb#321]
**Describe the bug**
I have a map that include keys that contain hyphens. DynamoDB throws error when I try to update the map values.
`\"ExpressionAttributeNames contains invalid key: Syntax error; key: \"#Cwxpw5TMn06X1-kscvMxS\"`

It seems that electrodb uses the map keys as the ExpressionAttribute keys without removing special characters or hashing the keys.

For example updating
```
attribute: {
  type: "map",
  properties: {
    'example-key XXX _ 1 2 3 4': { type: "string" },
  }
}
```
Outputs
```
"UpdateExpression": "SET #map.#example-key XXX _ 1 2 3 4 = :example-key XXX _ 1 2 3 4_u0, ...
```

**ElectroDB Version**
2.10.5

**ElectroDB Playground Link**
Use the [ElectroDB Playground](https://electrodb.fun/?#code/JYWwDg9gTgLgBAbzgUQHY2DAnnAvnAMyghDgCIBTAGwoGMZiATAIzIChaJUBneGAQ2Y04AXnJYIAVygB9AUIozU-EBXYcuvOBXSYcY1BQDuKXdgAUbOIivW4ICI2oAuG3bs6M2V2QHcA1txkADS27gBuFFDcwFw+AIwhYXbcUeHAtBQ+fv78YGDs7rih7vwwDMDMkjAU3K4IydbAjPWNdthgWeS8UMCoAOZJ7u5QFACOksCjLXAMkhRtxW0gea3D7VidPisFJetwYMSdsMC1a-vWAOQUAB4qYDQAtP4UOAAaH3AycPFwAExwADMcAALJd6rNNl0yD0+oM8Ht1rhFsklu4+k4bmc3MNDhAAFZ0GB1HHrMD+c4XAinKgzMjkoYXOCccAQGI1VwAbTIzTIAF02tY0cNhULbMKkPJhMiAJQAbjYis8elsADowGVaAALcxIZqua53cA0S54GVqxhlfjmcz8YJwCAy0QAPgdqtSMFtqp2nMN9yeL3en2+vwBwLBfPtl3C-Co80uMvN1lV-Qg5nlbCAA)

**Comment by tywalch** [comment]
Hi @metuuu 👋

First of all, how dare you. 

Second of all, this is a good find. 

In most cases the library strives limit the surface area of the validation and constraints it adds. This is typically because DynamoDB changes their own limits and constraints all the time, so it wouldn't be right for the library to try match/duplicate those. 

I'll note this change see about improving the formatting of these expression attributes/names 👍

Does DynamoDB even allow for spaces in their field names? I've never tried.

**Comment by metuuu** [comment]
My example was a bit over the top I agree 😄 I'm sure you don't have to worry about spaces.

Thank you!

**Comment by metuuu** [comment]
Just to note that in my actual case I was storing images to an object and the image IDs were generated with [nanoid](https://www.npmjs.com/package/nanoid) which can include hyphens and underscore. UUIDs would also have the same issue.
```js
images: {
  // { <image-id>: <s3-key> }
  type: CustomAttributeType<{ [imageId: string]: string }>>('any'),
}
```

**Comment by tywalch** [comment]
I just published the fix with `2.10.6`. You can use the same [playground link you created](https://github.com/tywalch/electrodb/issues/321#:~:text=Link%0AUse%20the-,ElectroDB%20Playground,-Owner) to see the change in action

We can close this ticket when you've verified your use case 👍

**Comment by metuuu** [comment]
All seems good! I agree with your solution, that if someone wants to shoot themselves in the leg by trying to store two items like `test-item` & `testitem` its completely their own fault.

Thank you 🙏🏻
